src/api/java/com/minecolonies/api/entity/CustomGoalSelector.java

```diff
@@ -139,12 +141,14 @@ public void addGoal(int priority, Goal task)
     @Override
     public void removeGoal(Goal task)
     {
-        this.availableGoals.stream().filter((goal) -> {
-            return goal.getGoal() == task;
-        }).filter(PrioritizedGoal::isRunning).forEach(PrioritizedGoal::stop);
-        this.availableGoals.removeIf((goal) -> {
-            return goal.getGoal() == task;
-        });
+        for(final PrioritizedGoal prioritizedGoal : new ArrayList<>(availableGoals))
+        {
+            if (prioritizedGoal.getGoal() == task)
+            {
+                prioritizedGoal.stop();
```

Comment:
This is a possible change of behaviour -- the previous code would `remove` as shown here but would only call `stop` when `isRunning`.

Having said that, it looks like `stop` will check `isRunning` internally anyway so it will probably work as expected as long as we trust that. 😄